--- ruler.c.@xbps	2024-06-29 09:12:15.765360217 -0400
+++ ruler.c.update	2026-06-06 05:46:24.297738124 -0400
@@ -19,6 +19,13 @@
 #include "asprintf.h"
 #include "ruler.h"
 
+static const char *atom_names[] = {
+	"WM_NAME",
+	"WM_CLASS",
+	"WM_WINDOW_ROLE",
+	"_NET_WM_WINDOW_TYPE"
+};
+
 extern FILE * yyin;
 
 struct list *last_d = NULL;
@@ -388,8 +395,10 @@
 			if (atom_name != NULL) {
 				if (i == 0)
 					sprintf(str, "%s", atom_name);
-				else
-					sprintf(str, "%s,%s", str, atom_name);
+				else {
+					size_t slen = strlen(str);
+					sprintf(str + slen, ",%s", atom_name);
+				}
 			}
 
 			free(atom_name);
@@ -902,7 +911,7 @@
 	if (parse_file(xdg_cfg_path) == 1 && no_of_configs == 0)
 		errx(1, "couldn't open config file '%s' (%s). No other config files supplied, exiting", xdg_cfg_path, strerror(errno));
 	free(xdg_cfg_path);
-	free(xdg_home);
+	// free(xdg_home);
 
 	for (i = 0; i < no_of_configs; i++) {
 		if (parse_file(configs[i]) != 0)
