--- peneira.kak	2024-07-03 18:21:22.782111139 -0400
+++ peneira.kak.update	2024-07-03 15:21:45.230165245 -0400
@@ -174,7 +174,7 @@
 
 # Configure highlighters and mappings
 define-command -hidden peneira-configure-buffer %{
-    remove-highlighter window/number-lines
+    # remove-highlighter window/number-lines
     add-highlighter buffer/peneira-matches ranges peneira_matches
     add-highlighter buffer/peneira-flag flag-lines @PeneiraFlag peneira_flag
     face window PrimaryCursor @PeneiraSelected
