--- reasymotion.kak	2025-11-09 15:17:13.220504169 -0500
+++ reasymotion.kak	2025-11-09 15:16:57.388502821 -0500
@@ -101,13 +101,13 @@
 
 def reasymotion-on-letter-to-word %{
     on-key %{
-        reasymotion-select-screen "s\b%val{key}\w*<ret>"
+        reasymotion-select-screen "s\b%val{key}<ret>"
     }
 }
 
 def reasymotion-on-letter-to-word-expand %{
     on-key %{
-        reasymotion-select-screen-expand "s\b%val{key}\w*<ret>"
+        reasymotion-select-screen-expand "s\b%val{key}<ret>"
     }
 }
 
